### DISCUSSION TITLE
how to store and read timestamps?

### DISCUSSION THREAD

**Discussion by guettli** [zombiezen/go-sqlite#71] [Q&A] [answered]
How to store and read timestamps?

I could use something like `time.Parse("2006-01-02 15:04:05.9999999", timestamp)` in my code, but I am unsure if this is the recommended approach.

**Comment by zombiezen** [comment, marked as answer]
It depends on what you need in your application; see #26 for some background.

The [built-in date & time functions](https://www.sqlite.org/lang_datefunc.html) will operate on a [variety of formats](https://www.sqlite.org/lang_datefunc.html#time_values) and Unix timestamps with millisecond precision. Sadly, [`DEFAULT current_timestamp`](https://sqlite.org/lang_createtable.html#the_default_clause) only gives second-level precision in the text format.
